omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" , "2.1.*" })
+public class IdentifierSourceResource extends DelegatingCrudResource<IdentifierSource>{
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		
+		DelegatingResourceDescription description = null;
+		
+		if (rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addSelfLink();
+			return description;
+		}
+		
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("identifierType");
+			
+			// links
+			description.addSelfLink();
+			if (rep instanceof DefaultRepresentation) {
+				description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			}
+		}
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getidentifierType(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType() + " - " + identifierSource.getName();
+	}
+	
+	@Override
+	public IdentifierSource newDelegate() {
+		return new SequentialIdentifierGenerator();
+	}
+	
+	@Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+		Object savedIdentifierSource = null;
+		String identifierSourceType = checkInput(postBody.get("sourceType"));
+		ArrayList<String> exception = new ArrayList<String>();
+		String identifierTypeUuid = checkInput(postBody.get("identifierType"));
+		PatientIdentifierType patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid);
+		String name = checkInput(postBody.get("name"));
+		String description = postBody.get("description").toString();
+		
+		if (StringUtils.isBlank(identifierTypeUuid) || patientIdentifierType == null) { 
+			exception.add("patient identifier type"); 
+			}
```

Comment:
Did you just forget to look into the indention problem?

Reply:
The indention problem is all over this class. Not just the above line.

Reply:
fixed